src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -70,7 +74,9 @@ internal sealed class ActiveSolutionBoundTracker : IActiveSolutionBoundTracker,
             this.gitEventsMonitor = gitEventsMonitor;
             this.logger = logger;
 
-            vsMonitorSelection = host.GetService<SVsShellMonitorSelection, IVsMonitorSelection>();
+            // TODO - MEF-ctor should be free-threaded -> we might be on a background thread ->
+            // calling serviceProvider.GetService is dangerous.
```

Comment:
Not sure how we'd fix this not-free-threaded issue, but I added this comment to call out the problem.